Tests/NetworkingTests/Mocks/MockFileDataWriter.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  FileDataWriterSpy.swift
+//
+//
+//  Created by Jan Kodeš on 24.01.2024.
+//
+
+import Foundation
+import Networking
+
+/// A test mock class for `FileDataWriting`.
+/// It writes into a file but let's us react when it's finished.
+class MockFileDataWriter: FileDataWriting {
+    var writeClosure: (() -> Void)?
+    private(set) var writeCalled = false
```

Comment:
why do we have properties writeCalled, receivedData or receivedURL when we don't use them anywhere or anyway?

Reply:
It's a practice that I have when writing any mocks, so that if needed anyone can use those properties to verify that it was called properly in the tests.

Reply:
When we have it can we also use it in tests? Aka write a test which  it uses. 

>  if needed anyone can use those properties to verify that it was called properly in the tests
I can write tons of code that anyone can potentially use in future, but usually until people know it they create their own solution anyway 🍶

Reply:
Added. I refactored the tests a bit and used the `MockFileWriter`.